contracts/Licences.sol

```diff
@@ -30,15 +31,18 @@ contract Licences is Ownable {
     mapping(bytes32 => bytes32) public _waterAccountIdToIdentifier;
     mapping(address => mapping(bytes32 => bytes32)) public _addressToZoneToWaterAccountId;
 
-    constructor() Ownable() {
+    OrderBook private _orderbook;
```

Comment:
```suggestion
    OrderBook immutable private _orderbook;
```

Reply:
LIES!

`TypeError: Immutable variables cannot have a non-value type.`

Reply:
Correction, you could put it there, but not in a few other places. I tried to use it too widely and made the compiler very unhappy.